pp.h

```diff
@@ -472,13 +488,15 @@ Does not use C<TARG>.  See also C<L</XPUSHu>>, C<L</mPUSHu>> and C<L</PUSHu>>.
 #define PUSHs(s)	(*++sp = (s))
 #define PUSHTARG	STMT_START { SvSETMAGIC(TARG); PUSHs(TARG); } STMT_END
 #define PUSHp(p,l)	STMT_START { sv_setpvn(TARG, (p), (l)); PUSHTARG; } STMT_END
+#define PUSHpvs(s)      PUSHp(""s"", sizeof(s)-1)
```

Comment:
D'oh, I just realised that that can't be used as an argument to a macro, so it would have to be written out as

```c
STMT_START { Perl_sv_setpvn(aTHX_ STR_WITH_LEN(s)); PUSHTARG; } STMT_END
```

Reply:
Ahyes, spaces added and force-pushed.

I thought about the `STR_WITH_LEN` macro but as you say, there's an annoying macro-nesting problem to it. :( The current implementation is a little copy-pastey on the sizeof part, but avoids copypasting the far more subtle body of each variation on the PUSHp macro.